lib/attrsets.nix

```diff
@@ -1798,6 +1798,43 @@ rec {
       then pkg.${output} or pkg.out or pkg
       else pkg;
 
+  /**
+    Like `getOutput` but with a list of fallback output names.
+    This function is alligned with `_overrideFirst()` from the `multiple-outputs.sh` setup hook.
+
+    # Inputs
+
+    `outputs`
+
+    : 1\. Function argument
+
+    `pkg`
+
+    : 2\. Function argument
+
+    # Type
+
+    ```
+    getOutput' :: [String] -> Derivation -> String
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.attrsets.getOutput'` usage example
+
+    ```nix
+    getOutput [ "include" "dev" ] pkgs.openssl
+    => "/nix/store/9rz8gxhzf8sw4kf2j2f1grr49w8zx5vj-openssl-1.0.1r-dev"
+    ```
+
+    :::
+  */
+  getOutput' = candidates: pkg:
+    let
+      output = builtins.head (builtins.filter (name: hasAttr name pkg) candidates ++ [ "out" ]);
```

Comment:
`getOutput` only tests for attrset membership, not for `elem output pkg.outputs`; I'm thinking the latter would've made more sense (especially in case of custom names like "include")

Could be more expensive to evaluate though